### PR TITLE
Diff indexing facet updates

### PR DESCRIPTION
**Pull request by dureuill**
This PR handles the fact that we are now given delta rather than the whole document to reindex in facet values.

This PR also removes `Index::faceted_documents_ids` and `Index::put_faceted_documents_ids` that were unused in the search.

### CODE REVIEW

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Awaiting the changes we talked about to support the deletion of the `facet_documents_ids` bitmap.

**Comment by dureuill** [comment]
PR is up to date.

File: milli/src/update/facet/bulk.rs [hunk 1]
```diff
@@ -133,20 +121,26 @@ impl<R: std::io::Read + std::io::Seek> FacetsUpdateBulkInner<R> {
...
         if self.db.is_empty(wtxn)? {
             let mut buffer = Vec::new();
             let mut database = self.db.iter_mut(wtxn)?.remap_types::<ByteSlice, ByteSlice>();
-            let mut cursor = new_data.into_cursor()?;
+            let mut cursor = delta_data.into_cursor()?;
             while let Some((key, value)) = cursor.move_on_next()? {
                 if !valid_lmdb_key(key) {
                     continue;
                 }
+                let value: KvReader<DelAdd> = KvReader::new(value);
```
Comment (Kerollmops) [inline milli/src/update/facet/bulk.rs:137]:
```suggestion
                let value = KvReaderDelAdd::new(value);
```

File: milli/src/update/facet/bulk.rs [hunk 2]
```diff
@@ -158,11 +152,14 @@ impl<R: std::io::Read + std::io::Seek> FacetsUpdateBulkInner<R> {
...
             let database = self.db.remap_types::<ByteSlice, ByteSlice>();
 
-            let mut cursor = new_data.into_cursor()?;
+            let mut cursor = delta_data.into_cursor()?;
             while let Some((key, value)) = cursor.move_on_next()? {
                 if !valid_lmdb_key(key) {
                     continue;
                 }
+
+                let value: KvReader<DelAdd> = KvReader::new(value);
```
Comment (Kerollmops) [inline milli/src/update/facet/bulk.rs:161]:
```suggestion
                let value = KvReaderDelAdd::new(value);
```

File: milli/src/update/facet/bulk.rs [hunk 3]
```diff
@@ -171,12 +168,15 @@ impl<R: std::io::Read + std::io::Seek> FacetsUpdateBulkInner<R> {
                 match database.get(wtxn, key)? {
                     Some(prev_value) => {
                         let old_bitmap = &prev_value[1..];
-                        CboRoaringBitmapCodec::merge_into(
-                            &[Cow::Borrowed(value), Cow::Borrowed(old_bitmap)],
-                            &mut buffer,
-                        )?;
+                        CboRoaringBitmapCodec::merge_deladd_into(value, old_bitmap, &mut buffer)?;
```
Comment (Kerollmops) [inline milli/src/update/facet/bulk.rs:171]: Is it correct to use `&prev_value[1..]`, here?

**Review by Kerollmops — COMMENTED** [review]
Looks good to me until we test it 😂

File: milli/src/update/facet/bulk.rs (hunk 3 above)
Comment (dureuill) [reply milli/src/update/facet/bulk.rs:171]: I think so:

1. The DB is `heed::Database<FacetGroupKeyCodec<ByteSliceRefCodec>, FacetGroupValueCodec>`.
2. `FacetGroupValueCodec` contains one byte of size (representing the level I think, starting at 1) followed by a roaring bitmap. So `prev_value[0]` is the size, and `prev_value[1..]` is the bitmap.

This is unchanged from the previous version of the code.

Comment (dureuill) [reply milli/src/update/facet/bulk.rs:170]: Do you want me to add some code comment here?

Comment (Kerollmops) [reply milli/src/update/facet/bulk.rs:170]: That's OK, I think, thank you. Indeed, a small comment could help a little bit!

Comment (dureuill) [reply milli/src/update/facet/bulk.rs:171]: Done

**Review by Kerollmops — APPROVED** [review]
Perfect 😃

**Pull request merged**